frontend/src/metabase/querying/notebook/components/NotebookDataPicker/NotebookDataPicker.tsx

```diff
@@ -203,29 +200,24 @@ function LegacyDataPicker({
   stageIndex,
   table,
   placeholder,
-  hasMetrics,
   isDisabled,
   onChange,
 }: LegacyDataPickerProps) {
-  const databaseId = Lib.databaseID(query);
   const tableInfo =
     table != null ? Lib.displayInfo(query, stageIndex, table) : undefined;
+  /**
+   * We try to blur the line between models and tables for embedding users.
+   * this property will change the way icons are displayed in the data picker trigger,
+   * so we need to remove it. Treating it as a table.
+   */
+  delete tableInfo?.isModel;
   const pickerInfo = table != null ? Lib.pickerInfo(query, table) : undefined;
-  const { data: card } = useGetCardQuery(
-    pickerInfo?.cardId != null ? { id: pickerInfo.cardId } : skipToken,
-  );
-  const context = useNotebookContext();
-  const modelList = getModelFilterList(context, hasMetrics);
 
   return (
-    <DataSourceSelector
+    <SimpleDataPicker
       key={pickerInfo?.tableId}
+      selectedEntity={pickerInfo?.tableId}
       isInitiallyOpen={!table}
-      selectedDatabaseId={databaseId}
-      selectedTableId={pickerInfo?.tableId}
-      selectedCollectionId={card?.collection_id}
-      databaseQuery={{ saved: true }}
-      canSelectMetric={modelList.includes("metric")}
```

Comment:
these props are used in different parts of the old data picker, and we don't need them for the new picker.

`selectedDatabaseId` is used in the database step
`selectedTableId` is used in the table step
`selectedCollectionId` is used in the entity picker when we select models
    ![image](https://github.com/user-attachments/assets/12683b16-c843-4306-abec-8eb01bcd7e0a)
`databaseQuery` is for the database step
`canSelectMetric` for the data bucket step
    ![image](https://github.com/user-attachments/assets/2c9f8039-fa98-4869-99b5-d2a509e4f87b)